File: #0791.custom-sort-string.cpp
class Solution {
public:
    string customSortString(string order, string s) {

        int size = s.size();
        map<char, int> m;
        for (int i = 0; i != size; i++) {
            m[s[i]] += 1;
        }
        
        size = order.size();
        string result = "";
        for (int i = 0; i != size; i++) {
            for (int j = 0; j != m[order[i]]; j++) {
                result += order[i];
            }
            m[order[i]] = 0;
        }

        for (auto it = m.begin(); it != m.end(); it++) {
            for (int j = 0; j != it -> second; j++) {
                result += it->first;
            }
        }
        return result;
    }
};

File: #0791.custom-sort-string.js
/**
 * @param {string} order
 * @param {string} s
 * @return {string}
 */
var customSortString = function(order, s) {
    let m = new Map();
    for (let i = 0; i < order.length; i++) {
        m.set(order[i], i);
    }

    let s2 = s.split("");
    s2.sort((left, right) => (m.get(left) || 0) - (m.get(right) || 0));
    return s2.join("");
};
